src/Sylius/Bundle/ApiBundle/Controller/UploadAvatarImageAction.php

```diff
@@ -13,45 +13,46 @@
 
 namespace Sylius\Bundle\ApiBundle\Controller;
 
-use ApiPlatform\Api\IriConverterInterface;
+use Sylius\Bundle\ApiBundle\Exception\AdminUserNotFoundException;
+use Sylius\Bundle\ApiBundle\Exception\NoFileUploadedException;
 use Sylius\Component\Core\Model\AdminUserInterface;
 use Sylius\Component\Core\Model\ImageInterface;
 use Sylius\Component\Core\Repository\AvatarImageRepositoryInterface;
 use Sylius\Component\Core\Uploader\ImageUploaderInterface;
 use Sylius\Component\Resource\Factory\FactoryInterface;
-use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
 use Symfony\Component\HttpFoundation\File\UploadedFile;
 use Symfony\Component\HttpFoundation\Request;
-use Webmozart\Assert\Assert;
 
 /** @experimental */
 final readonly class UploadAvatarImageAction
 {
     public function __construct(
         private FactoryInterface $avatarImageFactory,
         private AvatarImageRepositoryInterface $avatarImageRepository,
+        private RepositoryInterface $adminUserRepository,
         private ImageUploaderInterface $imageUploader,
-        private IriConverterInterface $iriConverter,
     ) {
     }
 
     public function __invoke(Request $request): ImageInterface
     {
+        /** @var AdminUserInterface $owner */
+        $owner = $this->adminUserRepository->find($request->attributes->get('id'));
```

Comment:
`attributes->get('id')` fallback is `null`, would be cool to check for value presence first.

Reply:
Ok, if `null` is passed, then we will get an error. So I decided to add an empty string as the default value, and then we won't get a user. #15498